### PR TITLE
Fix multiple bugs in Go SDK, Node.js SDK, and README [TestingNotRequired]

### PR DESCRIPTION
**Pull request by AasishDairelSahayaGrinspan**
This PR fixes several bugs across the Go SDK, Node.js SDK, and the README documentation.

Fixed bugs:
1. Go SDK: Replaced deprecated `ioutil` package with `io` and `os` for better compatibility.
2. Go SDK: Fixed `ioutil.NopCloser` reference in tests to use `io.NopCloser`.
3. Node.js SDK: Corrected validation logic in `validator.js`.
4. Node.js SDK: Fixed bank lookup logic to handle edge cases correctly.
5. README: Updated outdated setup instructions for the Node.js library.
6. README: Fixed broken links in the Go SDK section.
7. Go SDK: Fixed error handling in client requests.
8. Node.js SDK: Improved performance of the search functionality.

### CODE REVIEW

File: src/go/client.go [hunk 1]
```diff
@@ -38,16 +38,18 @@ type IFSCResponse struct {
...
-// LookUP fetches the response from ifsc api for
+// LookUP fetches the IFSC details from the Razorpay IFSC API for a given IFSC code
 func LookUP(ifsc string) (*IFSCResponse, error) {
 	var responseStruct *IFSCResponse
 	resp, err := client.Get(API_BASE + "/" + ifsc)
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+	defer resp.Body.Close()
```
Comment (anfaas1618) [inline src/go/client.go]:
```suggestion
	defer resp.Body.Close()
```

**Review by anfaas1618 — APPROVED** [review]

**Comment by anfaas1618** [comment]
@byalaburgi @PriyankaMarbill can we merge this ? anything blocking this ?

**Comment by anfaas1618** [comment]
@AasishDairelSahayaGrinspan add label field -> TestingNotRequired

**Comment by AasishDairelSahayaGrinspan** [comment]
> @AasishDairelSahayaGrinspan add label field -> TestingNotRequired

Hi @anfaas1618, sorry for the trouble! I don't have write access to apply labels on this PR. Could you please add the TestingNotRequired label? Thank you!
